Modules/arraymodule.c

```diff
@@ -130,11 +149,76 @@ enum machine_format_code {
 
 #define array_Check(op, state) PyObject_TypeCheck(op, state->ArrayType)
 
+static inline char *
+arraydata_safeitems(arraydata *data)
```

Comment:
```suggestion
static inline char *
array_items_ptr(arrayobject *self)
```

And then `array_items_ptr(self)` instead of `arraydata_safeitems(self->data)`.

Reply:
Changed all `->items` accesses to this whether needed or not.